src/core/connection.c

```diff
@@ -4772,6 +4772,27 @@ QuicConnRecvDatagramBatch(
                 }
             }
 
+            QUIC_ENCRYPT_LEVEL EncryptLevel = QuicKeyTypeToEncryptLevel(Packet->KeyType);
+            QUIC_PACKET_SPACE* PacketSpace = Connection->Packets[EncryptLevel];
+
+            if (PacketSpace) {
+                // FIXME: PacketSpace seems to be zero when receiving (coalesced)
+                // 0-RTT packets. This means they are currently not counted.
+                switch (QUIC_ECN_FROM_TOS(Datagrams[i]->TypeOfService)) {
+                    case QUIC_ECN_CE:
+                        PacketSpace->AckTracker.ReceivedECN.CE_Count++;
+                        break;
+                    case QUIC_ECN_ECT_1:
+                        PacketSpace->AckTracker.ReceivedECN.ECT_1_Count++;
+                        break;
+                    case QUIC_ECN_ECT_0:
+                        PacketSpace->AckTracker.ReceivedECN.ECT_0_Count++;
+                        break;
+                    case QUIC_ECN_NON_ECT:
+                        break;
+                }
+            }
```

Comment:
I'm also already setting this in the "ack packet" code path" in the ack_tracker.c.

Reply:
Oh yes